### ISSUE TITLE
etcdsnapshot pulls wrong rke-tools tag if custom system images is used 

### ISSUE THREAD

**Issue by kinarashah** [rancher/rke#1551]
```
system_images:
  alpine: "husseingalal/rke-tools:dev"
  nginx_proxy: "husseingalal/rke-tools:dev"
  cert_downloader: "husseingalal/rke-tools:dev"
  kubernetes_services_sidecar: "husseingalal/rke-tools:dev"
``` 
rke fails with: 
WARN[0097] [etcd] Can't pull Docker image [husseingalal/rke-tools:v0.1.40] on host [34.217.43.55]: Error response from daemon: manifest for husseingalal/rke-tools:v0.1.40 not found 
FATA[0097] [etcd] Failed to bring up Etcd Plane: Failed to pull image [husseingalal/rke-tools:v0.1.40] on host [34.217.43.55]: Error response from daemon: manifest for husseingalal/rke-tools:v0.1.40 not found 

it is still trying to get 0.1.40 tag.

**Comment by jiaqiluo** [comment]
Reproduce the bug in rke version v0.2.6

Use rke up to provision a cluster and see the following error in the output:

```
INFO[0060] [etcd] Pulling image [husseingalal/rke-tools:v0.1.34] on host [13.56.149.136]
FATA[0061] [etcd] Failed to bring up Etcd Plane: Can't pull Docker image [husseingalal/rke-tools:v0.1.34] for host [13.56.149.136]: Error response from daemon: manifest for husseingalal/rke-tools:v0.1.34 not found
```

**Comment by jiaqiluo** [comment]
The bug fix is validated in rke version v0.3.0-rc6

Use rke up to provision a cluster with custom system images and the cluster is provisioned successfully.
